text/0000-router-helpers.md

```diff
@@ -0,0 +1,442 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Router Helpers & Modifiers
+
+## Summary
+
+This RFC introduces new router helpers that represent a decomposition of functionality of what we commonly use `{{link-to}}` for. This RFC also deprecates the `{{link-to}}` component, `{{query-params}}` helper, and default query param serialization. Below is a list of the new helpers:
+
+```hbs
+{{transition-to}}
+{{url-for}}
+{{root-url}}
+{{is-active}}
+{{is-loading}}
+{{is-transitioning-in}}
+{{is-transitioning-out}}
+```
+
+This represents a super set of the functionality provided by [Ember Router Helpers](https://github.com/rwjblue/ember-router-helpers/) which has provided this RFC that confidence that a decomposition is possible.
+
+## Motivation
+
+`{{link-to}}` is the primary way for Ember applications to transition from route to route in your application. While this works for a lot of cases there are some use cases that are not well supported or supported at all by the framework. Below is an enumeration of cases that `{{link-to}}` does not address.
+
+### Anchor Tags
+
+We currently do not have a good solution for transitioning solely based on HTML anchors defined in the templating layer. For instance let say you are using [Ember Intl](https://github.com/ember-intl/ember-intl) to do internationalization for your application. Ember Intl uses the [ICU message format](http://userguide.icu-project.org/formatparse/messages) for the actual translation strings and supports having HTML within the string. Now lets say you want to put a link in a translation string and have it work like `{{link-to}}` works. In that case you either have to roll your own solution or use something like [Ember-href-to](https://github.com/intercom/ember-href-to). Another example where this would be useful is that links within markdown produced by addons like [Ember-CLI-Showdown](https://github.com/gcollazo/ember-cli-showdown) would just work. API's like `RouterService#transitionTo` can transition an application using relative URLs and we have an opportunity to leverage this functionality to support this use case.
```

Comment:
```suggestion
We currently do not have a good solution for transitioning solely based on HTML anchors defined in the templating layer.
For instance, let's say you are using [Ember Intl](https://github.com/ember-intl/ember-intl) to do internationalization for your application.
Ember Intl uses the [ICU message format](http://userguide.icu-project.org/formatparse/messages) for the actual translation strings and supports having HTML within the string.
Now, let's say you want to put a link in a translation string and have it work like `{{link-to}}` works.
In that case, you either have to roll your own solution or use something like [Ember-href-to](https://github.com/intercom/ember-href-to).
Another example where this would be useful is that links within markdown produced by addons like [Ember-CLI-Showdown](https://github.com/gcollazo/ember-cli-showdown) would just work.
APIs like `RouterService#transitionTo` can transition an application using relative URLs and we have an opportunity to leverage this functionality to support this use case.
```